TESTS/mbed_hal/lp_ticker/main.cpp

```diff
@@ -13,177 +13,93 @@
-     * to allow for hardware serial buffers to completely flush.
-
-     * This should be replaced with a better function that checks if the
-     * hardware buffers are empty. However, such an API does not exist now,
-     * so we'll use the wait_ms() function for now.
-     */
-    wait_ms(10);
-
-    ticker_set_handler(lp_ticker_data, cb_done_deepsleep);
-    ticker_remove_event(lp_ticker_data, &delay_event);
-    delay_ts = ticker_read(lp_ticker_data) + 1000000;
-
-    /*
-     * We use here the low power timer instead of microsecond timer for start and
-     * end because the microseconds timer might be disable during deepsleep.
-     */
-    lp_timer.reset();
-    lp_timer.start();
-    ticker_insert_event(lp_ticker_data, &delay_event, delay_ts, (uint32_t)&delay_event);
-    /* Make sure deepsleep is allowed, to go to deepsleep */
-    bool deep_sleep_allowed = sleep_manager_can_deep_sleep();
-    TEST_ASSERT_TRUE_MESSAGE(deep_sleep_allowed, "Deep sleep should be allowed");
-    sleep();
-    while (!complete);
-    lp_timer.stop();
-
-    TEST_ASSERT_UINT32_WITHIN(LONG_TIMEOUT, 1000000, complete_time);
-    TEST_ASSERT_TRUE(complete);
+    while (cycles--);
 }
 
-void lp_ticker_1s_sleep()
+/* Test that the ticker has the correct frequency and number of bits. */
+void lp_ticker_info_test()
 {
-    complete = false;
-    uint32_t delay_ts;
-
-    ticker_set_handler(lp_ticker_data, cb_done);
-    ticker_remove_event(lp_ticker_data, &delay_event);
-    delay_ts = ticker_read(lp_ticker_data) + 1000000;
-
-    sleep_manager_lock_deep_sleep();
-    timer.reset();
-    timer.start();
-    bool deep_sleep_allowed = sleep_manager_can_deep_sleep();
-    TEST_ASSERT_FALSE_MESSAGE(deep_sleep_allowed, "Deep sleep should be disallowed");
-    ticker_insert_event(lp_ticker_data, &delay_event, delay_ts, (uint32_t)&delay_event);
-    sleep();
-    while (!complete);
-    timer.stop();
-    sleep_manager_unlock_deep_sleep();
-
-    TEST_ASSERT_UINT32_WITHIN(LONG_TIMEOUT, 1000000, complete_time);
-    TEST_ASSERT_TRUE(complete);
-}
-#endif /* DEVICE_SLEEP */
+    const ticker_info_t* p_ticker_info = lp_ticker_get_info();
 
-void lp_ticker_500us(void)
-{
-    lp_ticker_delay_us(500, SHORT_TIMEOUT);
+    TEST_ASSERT(p_ticker_info->frequency >= 8000);
+    TEST_ASSERT(p_ticker_info->frequency <= 64000);
+    TEST_ASSERT(p_ticker_info->bits >= 12);
 }
 
-void lp_ticker_1ms(void)
+/* Test that the ticker continues operating in deep sleep mode. */
+void lp_ticker_deepsleep_test()
 {
-    lp_ticker_delay_us(1000, SHORT_TIMEOUT);
-}
+    intFlag = 0;
 
-void lp_ticker_1s(void)
-{
-    lp_ticker_delay_us(1000000, LONG_TIMEOUT);
-}
+    set_lp_ticker_irq_handler(ticker_event_handler_stub);
 
-void lp_ticker_5s(void)
-{
-    lp_ticker_delay_us(5000000, LONG_TIMEOUT);
+    lp_ticker_init();
+
+    /* Wait for green tea UART transmission before entering deep-sleep mode. */
+    wait_cycles(40000);
```

Comment:
Hi
Back on this new code: it seems that this 40000 value is not enough for STM32F401 for ex.
Why don't you use a simple wait(1) ?
Thx

Reply:
Hi,
What value would be enough for STM32F401? I'm not using `wait()` since it uses `ticker` mechanism which is tested here. Also original handler is replaced with our test handler.